### DISCUSSION TITLE
Can we disable css preload of dynamic components

### DISCUSSION THREAD

**Discussion by PlopTheReal** [sveltejs/kit#10044] [Q&A]
Let's say I have hundreds of dynamic components that are loaded dynamically with #await.
Is it possible to load the css of thoses when loading the component instead of loading all the css files on the initial page load?
Or inject the styles in the js chunk?

**Comment by teemingc** [comment]
See the part about the `preload` option in the server `handle` hook.
https://kit.svelte.dev/docs/hooks#server-hooks-handle
You can choose to only preload js as shown in the example below:
```ts
// src/hooks.server.ts
import type { Handle } from '@sveltejs/kit';

export const handle = (async ({ event, resolve }) => {
    const response = await resolve(event, {
        preload: ({ type }) => type === 'js'
    });

    return response;
}) satisfies Handle;
```
